pallets/fragments/src/lib.rs

```diff
@@ -474,6 +513,82 @@ pub mod pallet {
 			Ok(())
 		}
 
+		/// **Alters** the **custom metadata** of a **Fragment Definition** (whose ID is `fragment_hash`) by **adding or modifying a key-value pair** (`metadata_key.clone`,`blake2_256(&data.encode())`)
+		/// to the **BTreeMap field `custom_metadata`** of the **existing Fragment Definition's Struct Instance**.
+		/// Furthermore, this function also indexes `data` in the Blockchain's Database and stores it in the IPFS
+		///
+		/// # Arguments
+		///
+		/// * `origin` - The origin of the extrinsic / dispatchable function
+		/// * `fragment_hash` - **ID of the Fragment Definition**
+		/// * `metadata_key` - The key (of the key-value pair) that is added in the BTreeMap field `custom_metadata` of the existing Fragment Definition's Struct Instance
+		/// * `data` - The hash of `data` is used as the value (of the key-value pair) that is added in the BTreeMap field `custom_metadata` of the existing Fragment Definition's Struct Instance
+		#[pallet::weight(50_000)]
+		pub fn set_custom_metadata(
```

Comment:
I think that this needs to become `set_definition_metadata` as we might want more metadata (tbd)